pkgs/net-lib/net/server.rkt

```diff
@@ -0,0 +1,106 @@
+#lang racket/base
+
+(require racket/contract
+         racket/tcp)
+
+(provide
+ (contract-out
+  [start-server (server-proc/c (-> void?))]
+  [run-server (server-proc/c void?)]))
+
+(define (server-proc/c res/c)
+  (->* ((or/c #f string?)
+        listen-port-number?
+        (-> input-port? output-port? any))
+       (#:reuse? any/c
+        #:max-allow-wait exact-nonnegative-integer?
+        #:max-concurrent (or/c +inf.0 exact-positive-integer?)
+        #:listen-proc (-> listen-port-number? exact-nonnegative-integer? any/c (or/c #f string?) evt?)
+        #:accept-proc (-> any/c (values input-port? output-port?))
+        #:close-proc (-> any/c void?)
```

Comment:
I think we probably don't want to enforce that `close-proc` specifically return `void?` rather than `any/c`. Actually, it seems like `(-> any/c any)` would work fine.

Reply:
I prefer to enforce since typically these kinds of destructive procedures return void. We can relax it later if necessary, but I think it's more likely to catch issues as is.